### PR TITLE
Feature/generate email

### PR DESCRIPTION
**Pull request by nalinsb**
Create grid and save data

### CODE REVIEW

File: src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainees/page.tsx [hunk 1]
```diff
@@ -14,6 +14,35 @@ export function CartPageTrainees(props: CartPageTraineesProps) {
...
+  ]
+
+  const [data2, setData2] =useState([]);
+
+  useEffect(()=>{
+    getData();
+  },[])
+
+  const getData =() =>{
+    axios.get('http://localhost:7250/api/Trainee')
```
Comment (chan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainees/page.tsx:37]: take base url out

File: src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainers/page.tsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 import styles from './cart-page-trainers.module.css';
-import {Form} from 'antd'
+import {Form,Table} from 'antd'
 import Input from 'antd/es/input';
 import Button from 'antd/es/button';
-import {PlusOutlined} from '@ant-design/icons/lib/icons'
+import {AccountBookFilled, PlusOutlined} from '@ant-design/icons/lib/icons'
 import axios from 'axios';
 import React,{useState} from 'react';
+import { randomFill } from 'crypto';
```
Comment (chan4lk) [inline src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainers/page.tsx:9]: what is the package  'crypto' ?

File: src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Service/Class.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mail;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BlockchainCertificatesIssuer.API.Service
+{
+    internal class Class
```
[the remark this answers is not in the document]
Comment (nalinsb) [reply src/backend/BlockchainCertificatesIssuer/BlockchainCertificatesIssuer/Service/Class.cs:11]: Its a experiment , Now Done

File: src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/certificates/page.tsx [hunk 4]
```diff
@@ -7,6 +7,7 @@ import Input from 'antd/es/input';
 import axios from 'axios';
 import React,{useState} from 'react';
 import moment from 'moment';
+import trainers from  '../trainers/page'
```
[the remark this answers is not in the document]
Comment (nalinsb) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/certificates/page.tsx:10]: Ok, Done

File: src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainers/page.tsx (hunk 2 above)
Comment (nalinsb) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainers/page.tsx:9]: It is not use in the code, I removed.. Done

File: src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainees/page.tsx (hunk 1 above)
Comment (nalinsb) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/trainees/page.tsx:37]: Ok, Done

File: src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/courses/page.tsx [hunk 5]
```diff
@@ -38,18 +37,58 @@ function setStartDate(startdate, dateString) {
...
-    Description: description,
+    Title: titleval,
+    Details: description,
     StartDate: startdate,
     EndDate : enddate
    
   };
   console.log("abc",data)
-  const url ='';
+  const url ='http://localhost:7250/api/Course';
```
[the remark this answers is not in the document]
Comment (nalinsb) [reply src/portals/blockchain-frontend/apps/blockchain-frontend/app/(authenticated)/courses/page.tsx:91]: ok, Done

**Pull request merged**